planet/Decoration.ocd/Village.ocd/Crate.ocd/Script.c

```diff
@@ -2,8 +2,14 @@
 	@author Dustin Neß (dness.de)
 */
 
-protected func Construction()
+public func Definition(proplist def)
 {
-	//SetProperty("MeshTransformation", Trans_Mul(Trans_Rotate(RandomX(-180,180),RandomX(-10,10),10), Trans_Scale(RandomX(5,10))));
-	SetProperty("MeshTransformation", Trans_Mul(Trans_Rotate(RandomX(-180,180), 0, 1, 0), Trans_Scale(10)));
-}
\ No newline at end of file
+	def.MeshTransformation = Trans_Scale(10, 10, 100);
+}
+
+public func Construction() {
+	// Note: We apply the scaling after the rotation, which is the "wrong"
+	//       order. However, due to what I believe is an engine bug,
+	//       scaling down too much along the z axis makes the object disappear.
+	SetProperty("MeshTransformation", Trans_Mul(GetID().MeshTransformation, Trans_Rotate(RandomX(-180,180), 0, 1, 0)));
+}
```

Comment:
I think that the definition should just define the scale here, as in the changes to e.g. the candle. Why give all editor previews a fixed rotation if you change that to another random rotation once the object is placed in the game?

Reply:
> I think that the definition should just define the scale here, as in the changes to e.g. the candle.

It is already the same way?

> Why give all editor previews a fixed rotation if you change that to another random rotation once the object is placed in the game?

The original code has the random rotation. And as far as I can tell, it is not possible to have a random rotation in the preview.

Reply:
Ah, yes this one has scale only in the Definition() call. Why Trans_Scale(10, 10, 100) ? Should be Trans_Scale(10) only, imo.

Reply:
It's explained in the comment above. Scaling down too much along the z axis makes the object disappear.

Reply:
Then why was it 10 before? The Z axis should be the one that points towards the screen, so having this at 100 and x at 10 should create rectangular boxes in theory...

Reply:
At least for me, the crates didn't show up at all before.

And the scaling is applied in screen space, so it still looks like a cube.

Reply:
Still weird, but ok then!